x-pack/plugin/sql/qa/server/src/main/resources/conditionals.csv-spec

```diff
@@ -326,6 +326,23 @@ Pettey
 Heyers
 ;
 
+iifConditionWhere
+SELECT last_name FROM test_emp WHERE IIF(LENGTH(last_name) < 7, true, false) LIMIT 10;
```

Comment:
👍  for this test case. 
What about covering the simplest case (`WHERE boolField`) specifically too? The reason: the test case above gets translated into a script query, while the `WHERE boolField` into a terms query, and while there are plenty of test cases that test the `script` query, I don't think we have any that tests querying a `bool` field directly. Note: This would require adding a new field to the test dataset with `bool` type though.

Reply:
Yeh, I think it might make sense to add data for a new test index, with more types (like floats like bools) or extend one of the current ones. But not sure if part of this small PR. I'll open an issue to track it.